packages/fx-core/src/plugins/resource/apim/managers/questionManager.ts

```diff
@@ -30,23 +25,34 @@ export class VscQuestionManager implements IQuestionManager {
     this.apiVersionQuestion = apiVersionQuestion;
     this.newApiVersionQuestion = newApiVersionQuestion;
     this.existingOpenApiDocumentFunc = existingOpenApiDocumentFunc;
+    Object.setPrototypeOf(this, new.target.prototype);
   }
 
   async callFunc(func: Func, ctx: PluginContext): Promise<any> {
     throw BuildError(NotImplemented);
   }
 
-  async deploy(ctx: PluginContext, apimConfig: IApimPluginConfig): Promise<QTreeNode> {
+  async deploy(
+    projectPath: string,
+    envName: string,
+    apimState: PluginConfig | Json,
```

Comment:
May be the input param can be  `apimState` or `apimConfig`. Only one of them is needed.